test/jasmine/tests/mapbox_test.js

```diff
@@ -475,6 +492,22 @@ describe('mapbox plots', function() {
         });
     });
 
+    it('should be able to update the access token', function(done) {
+        var promise = Plotly.relayout(gd, 'mapbox.accesstoken', 'wont-work');
+
+        promise.catch(function(err) {
+            expect(gd._fullLayout.mapbox.accesstoken).toEqual('wont-work');
+            expect(err).toEqual(new Error(constants.mapOnErrorMsg));
```

Comment:
@chriddyp **Important** Setting a wrong access token currently:

- throws an error,
- clears the map, and
- does not update `fullLayout.mapbox.accesstoken`

Is this something you can work with the workspace? If not, please let me know. Alternatively, we could:

1. as the above, but clear `fullLayout.mapbox.accesstoken` if it is invalid
2. eat up the error ...
3. anything else?

Note also that setting an invalid `mapbox.style` does essentially the same.